### PR TITLE
FileManager updates

### PR DESCRIPTION
**Pull request by codeflight1**
| | |
|:---|:---|
|**_Originating Project/Creator_**| |
|**_Affected Component_**|Svc::FileManager|
|**_Affected Architectures(s)_**|  |
|**_Related Issue(s)_**|#297, #298|
|**_Has Unit Tests (y/n)_**|yes|
|**_Builds Without Errors (y/n)_**|yes|
|**_Unit Tests Pass (y/n)_**|yes|
|**_Documentation Included (y/n)_**|  |

---
## Change Description

This PR adds the FileSize command and ignoreErrors argument to the RemoveFile command in the FileManager component.

### CODE REVIEW

**Comment by LeStarch** [comment]
@timcanham will you review this?

File: Svc/FileManager/Commands.fppi [hunk 1]
```diff
@@ -36,3 +37,8 @@ async command AppendFile(
                           target: string size 256 @< The name of the file to append to
                         ) \
   opcode 0x05
+
+async command FileSize(
+                        fileName: string size 256 @< The file to get the size of
+                      ) \
```
Comment (timcanham) [inline Svc/FileManager/Commands.fppi:43]: To be consistent, I think we should add an `ignoreErrors` argument here as well, or remove them both and just make them warnings (my preference). Failing to remove a file or get a size is rarely an issue, but returning an error can complicate a long cleanup or listing activity.

**Review by timcanham — CHANGES_REQUESTED** [review]

File: Svc/FileManager/Commands.fppi (hunk 1 above)
Comment (codeflight1) [reply Svc/FileManager/Commands.fppi:43]: Should it then be included on the other commands (`CreateDirectory`, `MoveFile`, `RemoveDirectory`, possibly `ShellCommand`, and `AppendFile`) as well, or would it be better to make it a component-wide parameter?

File: Svc/FileManager/Commands.fppi [hunk 2]
```diff
@@ -20,6 +20,7 @@ async command RemoveDirectory(
 @ Remove a file
 async command RemoveFile(
                           fileName: string size 256 @< The file to remove
+                          ignoreErrors: bool @< Ignore non-existent files
```
Comment (LeStarch) [inline Svc/FileManager/Commands.fppi:23]: A suggestion here is to make this an enumeration.  That way it is a little more understandable in sequences.  E.g. a sequence:

```
fm.REMOVE_FILE /path/to/file TRUE
```
vs 
```
fm.REMOVE_FILE /path/to/file IGNORE_ERRORS
```
Using an enum self-describes without needing to look-up the command definition to remember which polarity the bool represented.

Comment (timcanham) [reply Svc/FileManager/Commands.fppi:23]: I agree with this approach. This one has gotten a little stale. @codeflight1: Are you willing to update this PR?

Comment (codeflight1) [reply Svc/FileManager/Commands.fppi:23]: I definitely can, but I was waiting for a decision on whether that should be added to each command or made a component-wide parameter (see other requested changes).

**Review by timcanham — APPROVED** [review]
The changes are good for a baseline for further considerations. OK to merge.

**Pull request merged**